indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1271,6 +1277,102 @@ public void remove(final Task task)
     }
   }
 
+  /**
+   * Acquire a read lock to perform the segment transactional append action for a given datasource.
+   * Also verifies that all the locks are of the type APPEND for the task.
+   * @param task task to perform the append action
+   * @param lockAcquireTimeoutMillis milliseconds to wait for lock acquisition
```

Comment:
What happens if the lock is not acquired within this time?

Reply:
Done